extensions/battleroyale/standalone.mjs

```diff
@@ -0,0 +1,532 @@
+          CellSet.fromObject({
+            name: LAYER_FIRE.layerName,
+            priority: LAYER_FIRE.zIndex,
+            cells: Array.from(this.fires),
+            layers: {[LAYER_FIRE.layerName]: 'BF'},
+            dynamic: true,
+          }),
+      );
+    }
+
+    this.cooldown = new Set();
+    this.fireCounter = 0;
+    this.gameStarted = false;
+    this.updateMapIntervalIDs = [];
+    this.participatePlayerIDs = new Set(); // store playerID
+    this.playerOldPosition = new Map(); // [string, mapCoord] store player's position before tp to game
+
+
+    await this.helper.gameState.registerOnPlayerUpdate((msg) => {
+      try {
+        this.onPlayerUpdate(msg);
+      } catch (e) {
+        console.error('battleroyale register player update listener failed: ', e, e.stack);
+      }
+    });
+
+    // register state functions for NPC
+    await this.helper.callS2sAPI('npc', 'registerStateFunc');
+    const initTime = new Date();
+    const startTime = new Date(Math.floor((+initTime + START_GAME_INTERVAL - 1) / START_GAME_INTERVAL) * START_GAME_INTERVAL + START_GAME_INTERVAL/2);
+    alarm(startTime, this.setGameInterval.bind(this));
+  }
+
+  /**
+   * Game interval
+   */
+  setGameInterval() {
+    alarm.recurring(START_GAME_INTERVAL, this.startGame.bind(this));
+  }
+
+  /**
+   * start Game
+   * @return {Boolean} success or not
+   */
+  async startGame() {
+    if (this.gameStarted) {
+      await this.resetGame();
+    }
+    if (this.participatePlayerIDs.size < 3) {
+      for (const playerID of this.participatePlayerIDs) {
+        await this.helper.callS2cAPI(playerID, 'notification', 'showNotification', 3000, '[battleroyale] player not enough to start (less than 3)');
+      }
+      return false;
+    }
+    for (const playerID of this.participatePlayerIDs) {
+      const playerloc = await this.helper.getPlayerLocation(playerID);
+      this.playerOldPosition.set(playerID, playerloc);
+      await this.helper.callS2cAPI(playerID, 'notification', 'showNotification', 3000, '[battleroyale] Game Start!');
+      await this.teleportPlayer(playerID);
+    }
+
+    this.gameStarted = true;
+    const updateMap = setInterval(async () => {
+      for (const mapName of this.arenaOfMaps.keys()) {
+        const needUpdate = this.updateEdge(
+            this.fires, this.arenaOfMaps.get(mapName), this.fireCounter,
+        );
+        if (!needUpdate) {
+          clearInterval(updateMap);
+          return;
+        }
+        await this.helper.broadcastCellSetUpdateToAllUser(
+            'update',
+            mapName,
+            CellSet.fromObject({
+              name: LAYER_FIRE.layerName,
+              cells: Array.from(this.fires),
+            }),
+        );
+        const players = this.helper.gameState.getPlayers();
+        players.forEach((player, playerID) => {
+          if (this.helper.gameMap.getCell(player.mapCoord, LAYER_FIRE.layerName)) {
+            this.killPlayer(playerID);
+          }
+        });
```

Comment:
The paramater `playerID` is the index of the array, not the playerID we are familiar with. Make sure that this place uses which one and rename the parameter if needed.

Reply:
The type of `players` is Map instead of Array. The code is intended.